### PR TITLE
Fixed #16327 -- Redirect "Save as new" to change view for the new object

### PR DESCRIPTION
**Pull request by alimony**
https://code.djangoproject.com/ticket/16327

### CODE REVIEW

**Comment by timgraham** [comment]
It looks like there aren't any tests for `save_as_continue`. Please review our [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist) for other missing items (e.g., release notes, `.. versionadded:: 1.10` annotation). After the patch is updated, uncheck "Patch needs improvement" on the ticket so it appears in the review queue.

**Comment by alimony** [comment]
Updated according to comments and patch checklist, but I would love some help with a last little thing, see comment: https://code.djangoproject.com/ticket/16327#comment:11

File: tests/admin_views/tests.py [hunk 1]
```diff
@@ -1119,9 +1119,30 @@ def setUp(self):
...
+    def test_save_as_continue_false(self):
+        """
+        Ensure saving a new object using "Save as new" redirects to changelist
+        instead of change view when the `save_as_continue` attribute is set to
+        `False`.
+        """
+        class CustomPersonAdmin(ModelAdmin):
+            save_as_continue = False
+
+        site.unregister(Person)
```
Comment (timgraham) [inline tests/admin_views/tests.py]: I tried a similar approach while working on acfaec3db5ba39de52f6e607e74343dccf72fba1 and came to the conclusion that this approach can't work (due to something like module caching). As far as I know, you'll have to register the admin to a separate `AdminSite`.

Comment (alimony) [inline tests/admin_views/tests.py]: I see, thanks for the help. Is something like `site3` along the other ones a proper way to do that? In your example you reuse the existing `site2` one, right?

Comment (timgraham) [inline tests/admin_views/tests.py]: I think there isn't much organization there. Using an existing site should be fine.

**Comment by alimony** [comment]
Thanks for helping me out there, test is fixed and passing now.

**Comment by timgraham** [comment]
merged in 08cd6a0e56bac04e322388e7059072ad98db303c, thanks!

**Pull request closed without merging**